scripts/plotting/cloud_regime_analysis.py

```diff
@@ -0,0 +1,1369 @@
+                    if ds_o[obs_ht_var_name][0] > ds_o[obs_ht_var_name][-1]: ht_selection = {obs_ht_var_name:slice(9999999999999,0)}
+                    else: ht_selection = {obs_ht_var_name:slice(0,9999999999999)}
+                    ds_o = ds_o.sel(tau_selection)
+                    ds_o = ds_o.sel(ht_selection)
+
+                    # Selcting only the relevant data and stacking it to shape n_histograms, n_tau * n_pc
+                    dims = list(ds_o.dims)
+                    dims.remove(obs_tau_var_name)
+                    dims.remove(obs_ht_var_name)
+                    histograms_o = ds_o.stack(spacetime=(dims), tau_ht=(obs_ht_var_name, obs_tau_var_name))
+                    weights_o = np.cos(np.deg2rad(histograms_o.lat.values)) # weights_o array to use with emd-kmeans
+
+                    # Turning into a numpy array for clustering
+                    mat_o = histograms_o.values
+
+                    # Removing all histograms with 1 or more nans in them
+                    indicies = np.arange(len(mat_o))
+                    is_valid = ~np.isnan(mat_o.mean(axis=1))
+                    is_valid = is_valid.astype(np.int32)
+                    valid_indicies_o = indicies[is_valid==1]
+                    mat_o=mat_o[valid_indicies_o]
+                    weights_o=weights_o[valid_indicies_o]
+
+                    if np.min(mat_o < 0):
+                        raise Exception (f'Found negative value in ds_o.{var_name}, if this is a fill value for missing data, convert to nans and try again')
+
+                    print(f'      -Fitting data') 
+
+                    # Compute cluster labels
+                    cluster_labels_temp_o = precomputed_clusters(mat_o, cl, wasserstein_or_euclidean, ds_o)
+
+                    # Taking the flattened cluster_labels_temp_o array, and turning it into a datarray the shape of obs_ds.var_name, and reinserting NaNs in place of missing data
+                    cluster_labels_o = np.full(len(indicies), np.nan, dtype=np.int32)
+                    cluster_labels_o[valid_indicies_o]=cluster_labels_temp_o
+                    cluster_labels_o = xr.DataArray(data=cluster_labels_o, coords={"spacetime":histograms_o.spacetime},dims=("spacetime") )
+                    cluster_labels_o = cluster_labels_o.unstack()
+
+                print(f'      -Plotting') 
+
+                plot_hists_obs(cl, cluster_labels, cluster_labels_o, ht_var_name, tau_var_name, adf)
+                plot_rfo_obs_base_diff(cluster_labels, cluster_labels_o, adf)
+
+            # Comparing to CAM baseline if not comparing to obs
+            else:
+                # path to h0 files
+                baseline_h0_data_path = adf.get_baseline_info("cam_hist_loc", required=True) + "/*h0*.nc"
+                # Time Range min and max, or None for all time
+                time_range_b = [str(adf.get_baseline_info("start_year")), str(adf.get_baseline_info("end_year"))] 
```

Comment:
I would replace this with the ADF-provided climo years dictionary:

```suggestion
                time_range_b = [str(adf.climo_yrs("syear_baseline")), str(adf.climo_yrs("eyear_baseline"))] 
```